src/script/common/c_converter.h

```diff
@@ -121,3 +121,11 @@ void                warn_if_field_exists(lua_State *L, int table,
 
 size_t write_array_slice_float(lua_State *L, int table_index, float *data,
 	v3u16 data_size, v3u16 slice_offset, v3u16 slice_size);
+
+// This must match the implementation in builtin/game/misc_s.lua
+inline lua_Number hash_node_position(v3s16 pos)
+{
+	return (((s64)pos.Z + 0x8000L) << 32)
```

Comment:
just a question, cannot we expose this directly instead of duplicating implementation on both side ? what is the cost ?

Reply:
The cost is that LuaJIT is unable to optimize this if the C++ func is used.

Reply:
In a later PR someone could look into calling the C++ implementation using LuaJIT's FFI library.